sdk/turing/router/config/router_config.py

```diff
@@ -206,16 +219,24 @@ def ensembler(self, ensembler: Union[RouterEnsemblerConfig, Dict]):
         if ensembler is None:
             # Init nop ensembler config if ensembler is not set
             self._ensembler = NopRouterEnsemblerConfig(final_response_route_id=self.default_route_id)
-        elif isinstance(ensembler, RouterEnsemblerConfig):
-            self._ensembler = ensembler
         elif isinstance(ensembler, dict):
             # Set fallback_response_route_id into standard ensembler config
             if ensembler["type"] == "standard" and "fallback_response_route_id" not in ensembler["standard_config"]:
                 ensembler["standard_config"]["fallback_response_route_id"] = self.default_route_id
             self._ensembler = RouterEnsemblerConfig(**ensembler)
         else:
             self._ensembler = ensembler
-
+        # Init child class types
+        if isinstance(self._ensembler, RouterEnsemblerConfig):
+            if self._ensembler.type == "nop" and not isinstance(self._ensembler, NopRouterEnsemblerConfig):
+                self._ensembler = NopRouterEnsemblerConfig.from_config(self._ensembler.nop_config)
+            if self._ensembler.type == "standard" and not isinstance(self._ensembler, StandardRouterEnsemblerConfig):
```

Comment:
Not sure if I'm missing something here, why is it 2 `if` conditions and then followed by `elif` here?

Reply:
Thanks, I missed updating the second check to an `elif`.